Cubical/CW/Connected.agda

```diff
@@ -0,0 +1,1319 @@
+{-# OPTIONS --safe --lossy-unification #-}
+
+{-
+This file contains the definition of an n-connected CW complex. This
+is defined by saying that it has non-trivial cells only in dimension
+≥n.
+
+The main result is packaged up in makeConnectedCW. This says that the
+usual notion of connectedness in terms of truncations (merely)
+coincides with the above definition for CW complexes.
+-}
+
+module Cubical.CW.Connected where
+
+open import Cubical.CW.Base
+open import Cubical.CW.Properties
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.GroupoidLaws
+open import Cubical.Foundations.Path
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Pointed
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+
+open import Cubical.Data.Nat renaming (_+_ to _+ℕ_)
+open import Cubical.Data.Nat.Order
+open import Cubical.Data.Nat.Order.Inductive
+open import Cubical.Data.Fin.Inductive.Base
+open import Cubical.Data.Fin.Inductive.Properties as Ind
+open import Cubical.Data.Sigma
+open import Cubical.Data.Sequence
+open import Cubical.Data.Unit
+open import Cubical.Data.Sum as ⊎
+open import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Bool
+
+open import Cubical.HITs.Sn
+open import Cubical.HITs.Pushout
+open import Cubical.HITs.Susp
+open import Cubical.HITs.SequentialColimit
+open import Cubical.HITs.SphereBouquet
+open import Cubical.HITs.PropositionalTruncation as PT
+open import Cubical.HITs.Truncation as TR
+open import Cubical.HITs.Wedge
+
+open import Cubical.Axiom.Choice
+open import Cubical.Relation.Nullary
+
+open import Cubical.Homotopy.Connected
+
+open Sequence
+
+private
+  variable
+    ℓ ℓ' ℓ'' : Level
+
+------ Definitions ------
+
+-- An n-connected CW complex has one 0-cell and no m<n-cells.
+yieldsConnectedCWskel : (A : ℕ → Type ℓ) (n : ℕ) → Type _
+yieldsConnectedCWskel A k =
+  Σ[ sk ∈ yieldsCWskel A ] ((sk .fst 0 ≡ 1) × ((n : ℕ) → n <ᵗ k → sk .fst (suc n) ≡ 0))
+
+-- Alternatively, we may say that its colimit is n-connected
+yieldsConnectedCWskel' : (A : ℕ → Type ℓ) (n : ℕ) → Type _
```

Comment:
It would be better, if this name differs not just by the apostrophe. Maybe you can use "combinatorial" or something like that as part of the name of the other one.